carbonmark-api/src/routes/projects/get.utils.ts

```diff
@@ -115,21 +128,27 @@ export const toGeoJSON = (
  * For polygon-bridged-carbon subgraph projects
  * Returns true if project has >=1 tonne in any pool
  * */
-export const isValidPoolProject = (
-  project: Pick<
-    CarbonOffset,
-    "balanceBCT" | "balanceNBO" | "balanceNCT" | "balanceUBO"
-  >
-) => {
-  const validProjects = [
-    project.balanceBCT,
-    project.balanceNCT,
-    project.balanceUBO,
-    project.balanceNBO,
-  ].filter((bal) => Number(bal) >= 1);
+export const isValidPoolProject = (project: FindQueryDigitalCarbon) => {
+  const validProjects = [];
+
+  for (const carbonCredit of project.carbonCredits) {
+    const poolBalances = carbonCredit.poolBalances;
+
+    for (const poolBalance of poolBalances) {
+      const poolAddresses = Object.values(POOL_INFO).map(
+        (pool) => pool.poolAddress
+      );
+      if (
+        poolAddresses.includes(poolBalance.pool.id) &&
+        Number(poolBalance.balance) > 0
+      ) {
+        validProjects.push(poolBalance);
+      }
+    }
+  }
```

Comment:
A couple of suggestions here:

* You can avoid the nested for loop by extracting all balances first to help with readability
* We only need to extract the poolAddresses once (so that can go outside the loop)
* You can avoid the loop entirely by using `some` (or `filter`)

```ts
export const isValidPoolProject = (project: FindQueryDigitalCarbon) => {
  const balances = project.carbonCredits.flatMap(extract("poolBalances"));
  const addresses = Object.values(POOL_INFO).map(extract("poolAddress"));

  return balances.some(
    (balance) =>
      addresses.includes(balance.pool.id) && Number(balance.balance) > 0
  );
};
```

This will help make things a little clearer and perform a little faster